### ISSUE TITLE
Project members not populated correctly on Adroit Mobile

### ISSUE THREAD

**Issue by grahammcculloch** [appdevdesigns/adroit-mobile#34]
See the example below from Shaun Futch. Notice that he is only listed as a member of one of the three projects returned. It seems that the `members` arrays of the projects are not being populated correctly. 

Affected endpoint: `/fcf_activities/mobile/myprojects`

```json
{
  "status": "success",
  "data": [
    {
      "IDProject": "FCFA",
      "ProjectName": "Family Connection Foundation Administration",
      "members": [
        {
          "IDPerson": 1321,
          "avatar": "/data/fcf/images/avatars/1321.jpg",
          "display_name": "Anthony Book"
        },
        {
          "IDPerson": 247,
          "avatar": "/data/fcf/images/avatars/0247.jpg",
          "display_name": "Bruce Rowe"
        },
        {
          "IDPerson": 74,
          "avatar": "/data/fcf/images/avatars/0074.jpg",
          "display_name": "Bryan Gerlach"
        },
        {
          "IDPerson": 1398,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Chutirat Panguan"
        },
        {
          "IDPerson": 24,
          "avatar": "/data/fcf/images/avatars/0024.jpg",
          "display_name": "Daniel Callahan"
        },
        {
          "IDPerson": 64,
          "avatar": "/data/fcf/images/avatars/0064.jpg",
          "display_name": "Darin Dunn"
        },
        {
          "IDPerson": 1433,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Deborah Stutzman"
        },
        {
          "IDPerson": 176,
          "avatar": "/data/fcf/images/avatars/0176.jpg",
          "display_name": "Imran Pasha"
        },
        {
          "IDPerson": 475,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Kanchana Somsikaew"
        },
        {
          "IDPerson": 1384,
          "avatar": "/data/fcf/images/avatars/1384.JPG",
          "display_name": "Mary Zing"
        },
        {
          "IDPerson": 230,
          "avatar": "/data/fcf/images/avatars/0230.jpg",
          "display_name": "Nathan Ullrich"
        },
        {
          "IDPerson": 464,
          "avatar": "/data/fcf/images/avatars/0464.jpg",
          "display_name": "Natthamon Siriprakaysaeng"
        },
        {
          "IDPerson": 715,
          "avatar": "/data/fcf/images/avatars/0715.jpg",
          "display_name": "Paul Meier"
        },
        {
          "IDPerson": 451,
          "avatar": "/data/fcf/images/avatars/0451.jpg",
          "display_name": "Pimpa Choochart"
        },
        {
          "IDPerson": 458,
          "avatar": "/data/fcf/images/avatars/0458.jpg",
          "display_name": "Sangdao Koocharoenthaworn"
        },
        {
          "IDPerson": 1400,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Sasithorn Baithong"
        },
        {
          "IDPerson": 1326,
          "avatar": "/data/fcf/images/avatars/1326.jpg",
          "display_name": "Shine Thomas"
        },
        {
          "IDPerson": 62,
          "avatar": "/data/fcf/images/avatars/0062.jpg",
          "display_name": "Timothy Dunham"
        },
        {
          "IDPerson": 1290,
          "avatar": "/data/fcf/images/avatars/1290.jpg",
          "display_name": "Utumporn Daungkaew"
        },
        {
          "IDPerson": 1025,
          "avatar": "/data/fcf/images/avatars/1025.jpg",
          "display_name": "Weerachai Somsak"
        }
      ],
      "teams": [ /* cut for brevity */ ]
    },
    {
      "IDProject": "LLW",
      "ProjectName": "Lahu Living Word (Theological Training)",
      "members": [
        {
          "IDPerson": 463,
          "avatar": "/data/fcf/images/avatars/0463.JPG",
          "display_name": "Chan Sakulcharoenlert"
        },
        {
          "IDPerson": 36,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Eric Christensen"
        },
        {
          "IDPerson": 466,
          "avatar": "/data/fcf/images/avatars/0466.JPG",
          "display_name": "Nutch Chongkoonakonchai"
        },
        {
          "IDPerson": 80,
          "avatar": "/data/fcf/images/avatars/0080.jpg",
          "display_name": "Shaun Futch"
        },
        {
          "IDPerson": 1305,
          "avatar": "/data/fcf/images/avatars/1305.jpg",
          "display_name": "Watthana Songkhunakornchai"
        }
      ],
      "teams": [ /* cut for brevity */ ]
    },
    {
      "IDProject": "GOHCC",
      "ProjectName": "TawSaeng Project",
      "members": [
        {
          "IDPerson": 11,
          "avatar": "/data/fcf/images/avatars/0011.jpg",
          "display_name": "Aaron Beckner"
        },
        {
          "IDPerson": 1331,
          "avatar": "/data/fcf/images/avatars/1331.jpg",
          "display_name": "Angela Highfield"
        },
        {
          "IDPerson": 1144,
          "avatar": "/data/fcf/images/avatars/1144.jpg",
          "display_name": "Bridget Blackburn"
        },
        {
          "IDPerson": 1448,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Cheryl Martin"
        },
        {
          "IDPerson": 1163,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Duane Danielsen"
        },
        {
          "IDPerson": 248,
          "avatar": "/data/fcf/images/avatars/0248.jpg",
          "display_name": "Joel Iyorwa"
        },
        {
          "IDPerson": 12,
          "avatar": "/data/fcf/images/avatars/0012.jpg",
          "display_name": "Karyn Beckner"
        },
        {
          "IDPerson": 1443,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Lisa Nalaschi"
        },
        {
          "IDPerson": 472,
          "avatar": "/data/fcf/images/avatars/0472.jpg",
          "display_name": "Maliwan Plempitiwongsa"
        },
        {
          "IDPerson": 1444,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Melissa Anson"
        },
        {
          "IDPerson": 1103,
          "avatar": "images/fcf_activities/icon_person_avatar.jpeg",
          "display_name": "Michael Feringa"
        },
        {
          "IDPerson": 1022,
          "avatar": "/data/fcf/images/avatars/1022.jpg",
          "display_name": "Ponrawin Udchoompisai"
        }
      ],
      "teams": [ /* cut for brevity */ ]
    }
  ]
}
```

**Comment by Hiro-Nakamura** [comment]
I didn't actually design this structure, but this is how I think it works:

There is a `tblMinistryTeamMembers` table that is used to define a person's connection to Projects or Ministries (teams).  They are specified separately.

In that table the `.codeServiceStatus` field specifies their current status with that project/ministry.  possible values `[ A, F, R, S, X]` 

- A : Applying
- F : Furlough
- R : Out But In  (???)
- S : Serving
- X : eXited

The service only reports him in projects where he is registered as `S` -> Serving

Shaun's entries (`IDPerson == 80`) look like this:
```
IDPerson  code  Project Ministry
80          S   LLW     -   
80          S   -       23      
80          S   -       13      
80          X   -       70  
80          X   FCFA    -
80          S   -       72  
80          S   -       53  
80          S   -       66      
80          S   -       113     
80          S   -       107     
80          S   -       14  
```
In his case, he is directly reported to be in 2 Projects: ` LLW` and `FCFA`.  And he is **only** currently Serving in `LLW`.  So he is reported in that Project's members, but not the others.

However, he is also Serving in 9 ministry teams (.`14` belongs to `GOHCC` which he doesn't have a Project link to).  The embedded team information for these projects should list him as a member there.

So I would go back to Anthony + Bruce and ask them if they need to update their Project information for Shaun.  Or clarify what is expected for us to report given the table strucutre and information.

**Comment by grahammcculloch** [comment]
Having discussed this, we have created a new endpoint at `/fcf_activities/mobile/myprojectmembers`. This returns the following structure (other fields omitted for brevity):

```json
{
  "projects": [
    {
      "teams": [
        {
          "memberIDs": [1]
        }
      ],
      "memberIDs": [1]
    }
  ],
  "members": [
    {
      "IDPerson": 1,
      "avatar": "",
      "display_name": ""
    }
  ]
}
```

The top-level `members` array contains a master list of all members in the same projects and/or the same teams as the authenticated user. Each project and team then contains a `memberIDs` array that lists the member IDs of members associated with that project/team. The front-end will have to convert this data to two lists for display in the Select component: Members in the specified team and Members that are in the project the specified team belongs to. This will also make it more like the web app UI for selecting people to tag.

**Comment by grahammcculloch** [comment]
Fixed with 1.2.0 v6 (Android) and 1.2.0 v7 (iOS)
